presto-main/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java

```diff
@@ -63,6 +67,31 @@ public static long valueAtQuantileBigint(@SqlType("qdigest(bigint)") Slice input
         return new QuantileDigest(input).getQuantile(quantile);
     }
 
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double quantileAtValueDouble(@SqlType("qdigest(double)") Slice input, @SqlType(StandardTypes.DOUBLE) double value) {
+        return quantileAtValueBigint(input, doubleToSortableLong(value));
+    }
+
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double quantileAtValueReal(@SqlType("qdigest(real)") Slice input, @SqlType(StandardTypes.REAL) long value) {
+        return quantileAtValueBigint(input, floatToSortableInt(intBitsToFloat((int) value)));
+    }
+
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double quantileAtValueBigint(@SqlType("qdigest(bigint)") Slice input, @SqlType(StandardTypes.BIGINT) long value) {
+        QuantileDigest digest = new QuantileDigest(input);
+        // when qdigest is empty, return the upper bound quantile to be consistent with non-empty scenarios
```

Comment:
For non-empty scenarios, suppose there are 100 values with 40 of them smaller than value X, then the quantile number returned is 0.4

I think maybe it is better to set empty scenario to return 0.0 instead of 1.0? because for non-empty scenario, if no values in the digest is smaller than X, it will return 0.0

Reply:
Yes, return 0 for empty qdigest makes more sense.

Could you combine to a single commit for better readability in commit history?